packages/use-reapit-data/src/utils.ts

```diff
@@ -0,0 +1,113 @@
+  return accessToken
+    ? {
+        Authorization: `Bearer ${accessToken}`,
+        'api-version': 'latest',
+        'Content-Type': 'application/json',
+        ...headers,
+      }
+    : null
+}
+
+export const handleReapitError = (error: AxiosError<any>, defaultMessage?: string): string => {
+  const { message } = error
+  const reapitError: ReapitError = error?.response?.data
+  const { description, errors } = reapitError ?? {}
+  const messageString = description
+    ? description
+    : message
+    ? message
+    : defaultMessage
+    ? defaultMessage
+    : 'An unknown error has occurred, please refresh the page and try again.'
+  const fieldErrors = errors?.map(({ field, message }) => `"${field}: ${message}"`)
+  const fieldString = fieldErrors ? fieldErrors.join(', ') : ''
+
+  return `${messageString} ${fieldString}`
+}
+
+export const getUrl = (action: GetAction | UpdateAction, queryParams?: Object, uriParams?: Object) => (): string => {
+  const { api, path } = action
+  const deSerialisedPath = uriParams
+    ? Object.keys(uriParams).reduce<string>((path, uriReplaceKey) => {
+        return path.replace(`{${uriReplaceKey}}`, uriParams[uriReplaceKey])
+      }, path)
+    : path
+  const query = qs.stringify(queryParams, { encode: false })
+  const url = `${api}${deSerialisedPath}${query ? `?${query}` : ''}`
+  return url
+}
+
+/** Util for useReapitGet queryParams to handle batch ids
+ *
+ * Problem: useReapitGet generates a query from an object so dupicate keys are not possible. The API sometimes requires
+ * us to use multiple queries of the same key eg ?foo=bar&foo=baz&foo=bat
+ *
+ * This util takes a given list and outputs as a string that can be used as the value in a useReapitGet queryParams call
+ * Usage: 
+ * 
+  const [trafficEvents] = useReapitGet<TrafficEventsModel>({
+    reapitConnectBrowserSession,
+    action: getActions(window.reapit.config.appEnv)[GetActionNames.getTrafficStats],
+    queryParams: {
+      appId: listToBatchQuery<AppSummaryModel>(apps, 'id', 'appId')
+    },
+    fetchWhenTrue: [dateFrom, dateTo],
+  })
+
+   listToBatchQuery<AppSummaryModel>(apps, 'id', 'appId') === MOCK_ID1&appId=MOCK_ID2&appId=MOCK_ID3
+ * 
+ */
+
+export const listToBatchQuery = <ListType>(list: ListType[], listKey: keyof ListType, queryKey: string): string =>
+  list.reduce((query: string, nextItem: ListType, index: number) => {
+    const listValue = String(nextItem[listKey])
+    if (!index) {
+      return listValue
+    }
+    return `${query}&${queryKey}=${listValue}`
+  }, '')
+
+// Same principle for this function but where your list is an array of strings or numbers
+export const stringListToBatchQuery = (list: (string | number)[], queryKey: string): string =>
+  list.reduce((query: string, nextItem: string | number, index: number) => {
+    if (!index) {
+      return String(nextItem)
+    }
+    return `${query}&${queryKey}=${nextItem}`
+  }, '')
```

Comment:
is this an issue with how .NET handles array query params? Does `key[0]=&key[1]=` not work? I remember having this problem but not how I solved it

Reply:
Yeah, so I spoke to Ryan about this. We both thought it should work fine but he couldn't guarantee it. It was a just in case util, I copied across from React Utils rather than a new thing. It's used in a few places, so didn't want to bin it and I want to be able to remove the old hooks from the React Utils package ultimately.

Going to leave in for now but we may well deprecate later